### PR TITLE
feat(request): fix post request validation, upgrade dependencies, rem…

### PR DESCRIPTION
**Pull request by blacksonic**
…ove duplication from headers

### CODE REVIEW

File: requestOption.js [hunk 1]
```diff
@@ -38,7 +38,18 @@ SuiteRequestOption.prototype = {
...
+    this.headers.forEach(function(addedHeader) {
+      if (addedHeader[0] === header[0]) {
+        addedHeader[1] = header[1];
+        exists = true;
+      }
+    });
+
+    if (!exists) {
+      this.headers.push(header);
+    }
```
Comment (drawain) [inline requestOption.js]: Couldn't it be a simple reduce method? Push all headers to the accumulator except if it exists.

**Comment by Valetudox** [comment]
Pls write some test (requestOption.spec.js)

File: requestOption.js (hunk 1 above)
Comment (Valetudox) [inline requestOption.js]: +1 
Hard too understand

**Comment by blacksonic** [comment]
Has test for it, it jus seeable in the commit not here

File: requestOption.js (hunk 1 above)
Comment (blacksonic) [inline requestOption.js]: find it and if not exists just push?

**Comment by gabor-balla-emarsys** [comment]
OMG a break; statement

**Comment by gabor-balla-emarsys** [comment]
I would still prefer not to mutate the original headers array.
Why moving expect to global namespace?
Other than these, it seems OK to me.

**Comment by blacksonic** [comment]
expect in global removes necessary require of chai everywhere

**Pull request merged**